source/containerimage/pull.go

```diff
@@ -338,8 +421,8 @@ func (p *puller) Snapshot(ctx context.Context, g session.Group) (ir cache.Immuta
 		}
 	}
 
-	if p.id.RecordType != "" && current.GetRecordType() == "" {
-		if err := current.SetRecordType(p.id.RecordType); err != nil {
+	if p.RecordType != "" && current.GetRecordType() == "" {
+		if err := current.SetRecordType(p.RecordType); err != nil {
 			return nil, err
 		}
 	}
```

Comment:
I think in case of oci layout you might want to `Unlazy()` the ref in here right away. The lazy semantics that are useful for the registry are not that useful in here and may cause issues if the session drops.

Reply:
I don't actually get how these work. What is `RecordType` and how does it implement lazy pulls?

Reply:
RecordType is unrelated here.  @sipsma What is the simplest way to trigger unlazy() for `current` in here? Looks like we didn't add a `ref.Unlazy()` method.

Reply:
@sipsma PTAL

@deitch Not a blocker for this PR. We can do this in follow-up.